rp2040-hal/src/adc.rs

```diff
@@ -1,3 +1,104 @@
 //! Analog-Digital Converter (ADC)
 // See [Chapter 4 Section 9](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
 // TODO
+
+use hal::adc::{Channel, OneShot};
+use pac::{ADC, RESETS};
+
+use crate::{
+    gpio::Pin,
+    gpio::{
+        bank0::{Gpio26, Gpio27, Gpio28, Gpio29},
+        FloatingInput,
+    },
+    resets::SubsystemReset,
+};
+
+/// Adc
+pub struct Adc {
+    device: ADC,
+}
+
+impl Adc {
+    /// Create new adc struct and bring up adc
+    pub fn new(device: ADC, resets: &mut RESETS) -> Self {
+        device.reset_bring_down(resets);
+        device.reset_bring_up(resets);
+
+        // Enable adc and temp sensor
```

Comment:
This isn't enabling the temp sensor, could you reduce this comment to just "Enable adc"
I know the section in the manual is "ADC and temperature sensor", but it's really two separate components and you're only dealing with one of them here.

Reply:
Fixed